crates/apollo-mcp-server/src/runtime/trace.rs

```diff
@@ -0,0 +1,111 @@
+use opentelemetry::{KeyValue, global, trace::TracerProvider as _};
+use opentelemetry_sdk::{
+    Resource,
+    metrics::{MeterProviderBuilder, PeriodicReader, SdkMeterProvider},
+    trace::{RandomIdGenerator, SdkTracerProvider},
+};
+use opentelemetry_semantic_conventions::{
+    SCHEMA_URL,
+    attribute::{DEPLOYMENT_ENVIRONMENT_NAME, SERVICE_VERSION},
+};
+use tracing_opentelemetry::{MetricsLayer, OpenTelemetryLayer};
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+use crate::runtime::{Config, logging::Logging};
+
+// Create a Resource that captures information about the entity for which telemetry is recorded.
+fn resource() -> Resource {
+    Resource::builder()
+        .with_service_name(env!("CARGO_PKG_NAME"))
+        .with_schema_url(
+            [
+                KeyValue::new(SERVICE_VERSION, env!("CARGO_PKG_VERSION")),
+                KeyValue::new(
+                    DEPLOYMENT_ENVIRONMENT_NAME,
+                    std::env::var("ENVIRONMENT").unwrap_or_else(|_| "develop".to_string()),
```

Comment:
This should more conventional in most ecosystems like Docker, Node.js, Rails, etc. (Not confident about Rust though 😅)

```suggestion
                    std::env::var("ENVIRONMENT").unwrap_or_else(|_| "development".to_string()),
```